torch/csrc/jit/codegen/cuda/transform_iter.cpp

```diff
@@ -440,6 +473,282 @@ int BestEffortReplay::findFirstMismatchedID(
+          }
+        }
+
+        // Set up compliment map
+        for (auto forwarded_id : forwarded_ids) {
+          compliment_map.emplace(std::make_pair(forwarded_id, compliment_ids));
+        }
+      }
+    }
+  }
+};
+
+} // namespace
+
+BestEffortReplay BestEffortReplay::replayCasP(
+    const TensorView* consumer,
+    const TensorView* producer,
+    int producer_compute_at_axis,
+    const RootDomainMap& root_map) {
+  if (producer_compute_at_axis < 0)
+    producer_compute_at_axis += (int)producer->nDims() + 1;
+
+  TORCH_INTERNAL_ASSERT(
+      producer_compute_at_axis >= 0 &&
+          (unsigned int)producer_compute_at_axis <= producer->nDims(),
+      "Invalid axis provided to BestEffortReplay::replayCasP.");
+
+  // Copied from TransformReplay::replayCasP
+  // TODO: Remove duplicate code
+
+  // producer ids we need to match in consumer
+  std::vector<IterDomain*> producer_CA_ids(
+      producer->domain()->domain().begin(),
+      producer->domain()->domain().begin() + producer_compute_at_axis);
+  producer_CA_ids = TensorDomain::noReductions(producer_CA_ids);
+
+  // If producer has an rfactor root, that's what will match the consumer
+  std::vector<IterDomain*> producer_root = producer->getMaybeRFactorDomain();
+
+  // Figure out all inputs required to generate the compute_at dimensions. We
+  // need all deps because inputs on producer may be in getRootDomain, but we
+  // may need in rFactorDomain
+  auto all_CA_id_deps = DependencyCheck::getAllValsBetween(
+      {producer_root.begin(), producer_root.end()},
+      {producer_CA_ids.begin(), producer_CA_ids.end()});
+
+  // Figure out minimal set of root IDs needed to produce producer_CA_ids:
+  std::unordered_set<IterDomain*> producer_CA_root_ids;
+  for (IterDomain* id : producer_root) {
+    if (std::find(all_CA_id_deps.begin(), all_CA_id_deps.end(), id) !=
+        all_CA_id_deps.end()) {
+      producer_CA_root_ids.emplace(id);
+    }
+  }
+
+  const auto p2c_root_map = root_map.mapProducerToConsumer(
+      producer->domain(), consumer->domain(), producer_CA_root_ids);
+
+  ConsumerForwardingInfo consumer_forwarding_info(producer, consumer);
+
+  auto consumer_replay = BestEffortReplay(
+      consumer->domain()->domain(),
+      producer_CA_ids,
+      p2c_root_map,
+      consumer_forwarding_info.forwarding_map);
+
+  // Need to adjust leaf map based on forwarding before returning.
+
+  // ID's could go through more than one forward iteration in the map before it
+  // terminates. Grab every id between the forwarded id, and what it was
+  // forwarded to
+  std::vector<IterDomain*> expanded_forwarded_ids;
+  for (auto forwarded_id : consumer_replay.forwarded_ids_) {
+    auto id = forwarded_id;
+    if (consumer_forwarding_info.forwarding_map.find(id) !=
+        consumer_forwarding_info.forwarding_map.end()) {
+      expanded_forwarded_ids.emplace_back(id);
+      id = consumer_forwarding_info.forwarding_map.at(id);
+    } else {
+      continue;
+    }
```

Comment:
This block seems to be just no-op. `expanded_forwarded_ids` is created, but it's just a local vector. 

Should it be just removed?